synapse/storage/data_stores/main/event_push_actions.py

```diff
@@ -901,14 +915,14 @@ def _rotate_notifs_before_txn(self, txn, rotate_to_stream_ordering):
             table="event_push_summary",
             values=[
                 {
-                    "user_id": key[0],
-                    "room_id": key[1],
-                    "notif_count": summary["notif_count"],
-                    "unread_count": summary["unread_count"],
-                    "stream_ordering": summary["stream_ordering"],
+                    "user_id": summary.user_id,
+                    "room_id": summary.room_id,
+                    "notif_count": summary.notif_count,
+                    "unread_count": summary.unread_count,
+                    "stream_ordering": summary.stream_ordering,
                 }
-                for key, summary in summaries.items()
-                if summary["old_user_id"] is None
+                for summary in summaries.values()
```

Comment:
maybe you could use `for ((user_id, room_id), summary) in summaries.items` to save putting the user_id and room_id in the `EventPushSummary`? shrug